### PR TITLE
Megapack of golang parsers

### PR DESCRIPTION
**Pull request by timgluz**
Hi, 

this PR adds support for most popular Golang pkg-managers(`Dep`, `Glide`, `Go Vendor`) and updates outdated implementation of `GodepParser`.

Those parsers are currently used in `GodepsGithubCrawler`.

As Godep parser doesnt have yet proper implemention of `parse_requested_version`, then those
parsers are not yet added into `ParserStrategy` - will be solved in another PR.

### CODE REVIEW

File: lib/versioneye/models/version.rb [hunk 1]
```diff
@@ -66,6 +66,18 @@ def security_vulnerabilities
     SecurityVulnerability.where(:_id.in => sv_ids)
   end
 
+  # it adds project dependency into project and updates some metrics
+  def add_dependency_to_project(project, dep_db, prod_db)
```
Comment (reiz) [inline lib/versioneye/models/version.rb]: @timgluz What has this method to do with versions? Somehow version.rb looks like the wrong place for this method.

File: lib/versioneye/parsers/common_parser.rb [hunk 2]
```diff
@@ -89,7 +111,7 @@ def fetch_response url
 
   def fetch_response_body( url )
     response = self.fetch_response( url )
-    response.body
+    response.body if response
```
Comment (reiz) [inline lib/versioneye/parsers/common_parser.rb]: And what happens if not? The return value for the case that the response is nil should be clear.

File: lib/versioneye/parsers/common_parser.rb [hunk 3]
```diff
@@ -244,5 +266,16 @@ def self.mix_file?(filename)
 
     return false
   end
+
+  def add_dependency_to_project(project, dep_db, prod_db)
```
Comment (reiz) [inline lib/versioneye/parsers/common_parser.rb]: @timgluz This method is already in the `version.rb` class. Right? Somehow it looks redundant.

**Comment by reiz** [comment]
@timgluz Overall it looks good. I added some comments to the code but some things look a bit strange to me. If they get resolved I will merge this into the master branch.

File: lib/versioneye/models/version.rb (hunk 1 above)
Comment (timgluz) [inline lib/versioneye/models/version.rb]: good catch, not sure how the method from `CommonParser` got into that model;

File: lib/versioneye/parsers/common_parser.rb (hunk 2 above)
Comment (timgluz) [inline lib/versioneye/parsers/common_parser.rb]: Well, it's handled by the `rescue` block which doesnt pass silently when request failed; it makes debugging easier;

File: lib/versioneye/parsers/common_parser.rb (hunk 3 above)
Comment (timgluz) [inline lib/versioneye/parsers/common_parser.rb]: yep,  it was some accident on editor;

**Comment by timgluz** [comment]
Ok, i will fix it and merge newest changes from master

File: lib/versioneye/models/version.rb (hunk 1 above)
Comment (reiz) [inline lib/versioneye/models/version.rb]: @timgluz Looks like random method hopping :-)

**Pull request merged**